clappr/clappr-ios
# Decouple Fullscreen logic of Core and MediaControl

## Summary
**OBS: Changes only for iOS.**

At the moment the complexity of fullscreen logic is spread between Core and MediaControl.

When the option ``kFullscreenByApp: true`` means that the Clappr doesn't set the player to fullscreen when the user taps on the fullscreen button, instead it triggers an event to notify the desire of the user. Then the app takes the responsibility to put the player in the fullscreen mode.

The Core and MediaControl class hold the logic to know when its necessary to only trigger event, change the icon and change the layout state to embed or fullscreen.

## Goal
Simplify and decouple the logic of fullscreen.
### MediaControl
Media controls don't hold any logic of fullscreen controlled by the app.

#### When user taps on fullscreen button the MediaControl triggers the event: 
 - userRequestExitFullscreen
 - userRequestEnterInFullscreen
#### When core triggers didExitFullscreen/didEnterFullscreen
 - Changes the icon to the respective state

### Core
Now Core holds both logic of fullscreen: Controlled by App or by the Player.

## How to test
1. Run all unit tests.
2. Run the Example app and check if the icon of fullscreen is being set correctly.
3. On ``ViewController.swift``, append the option ``kFullscreen: true`` and check if the player starts in fullscreen mode.
